### PR TITLE
Vecsim normalize

### PR DESCRIPTION
**Pull request by GuyAv46**
exported normalization function and removed normalization step from `getDistanceFrom` function

### CODE REVIEW

File: src/VecSim/vec_sim.cpp [hunk 1]
```diff
@@ -46,6 +46,12 @@ extern "C" double VecSimIndex_GetDistanceFrom(VecSimIndex *index, size_t id, con
     return index->getDistanceFrom(id, blob);
 }
 
+extern "C" void VecSim_Normalize(void *blob, size_t dim, VecSimType type) {
```
Comment (DvirDukhan) [inline src/VecSim/vec_sim.cpp:49]: do you assume the buffer is mutable?

File: tests/unit/test_bruteforce.cpp [hunk 2]
```diff
@@ -971,6 +971,8 @@ TEST_F(BruteForceTest, brute_get_distance) {
     }
 
     void *query = v1;
+    void *norm = v2;
```
Comment (DvirDukhan) [inline tests/unit/test_bruteforce.cpp:974]: please state it is a unit vector. maybe change it to {1, 1,}

File: src/VecSim/vec_sim.cpp (hunk 1 above)
Comment (GuyAv46) [reply src/VecSim/vec_sim.cpp:49]: Yes. either send the original vector or make a copy yourself

Comment (GuyAv46) [reply src/VecSim/vec_sim.cpp:49]: although I can move `VecSimType_sizeof` to the library and then with the same args have the copy part also

File: src/VecSim/vec_sim.h [hunk 3]
```diff
@@ -49,12 +49,21 @@ int VecSimIndex_DeleteVector(VecSimIndex *index, size_t id);
  * metric.
  * @param id the id of the vector in the index.
  * @param blob binary representation of the second vector. Blob size should match the index data
- * type and dimension.
+ * type and dimension, and pre-normalized if needed.
```
Comment (alonre24) [inline src/VecSim/vec_sim.h:52]: `and pre-normalized if needed.`
This should be more emphasised in my opinion, as it is not trivial...

**Review by alonre24 — DISMISSED** [review]
Much better and cleaner... nice job!

**Review by alonre24 — APPROVED** [review]

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**